src/transformers/modeling_xlnet.py

```diff
@@ -1091,7 +1091,11 @@ def forward(
             output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
         )
         return_dict = return_dict if return_dict is not None else self.config.use_return_dict
-        use_cache = self.training or (use_cache if use_cache is not None else self.config.use_cache)
+
+        if self.training:
+            use_mems = use_mems if use_mems is not None else self.config.use_mems_train
+        else:
+            use_mems = use_mems if use_mems is not None else self.config.use_mems_eval
```

Comment:
Nit: I'd personally prefer in two lines:
```
default_use_mems = self.config.use_mems_train if self.training else self.config.use_mems_eval
use_mems = use_mems if use_mems is not None else default_use_mems
```
but it's a matter of personal taste, probably.